### PR TITLE
Create rancher service for internal communication to prevent loopback through load balancer

### PR DESCRIPTION
**Pull request by Oats87**
There are certain cases (when running Rancher in a docker container) where we do not want Rancher to attempt to connect via the exposing load balancer. This PR will create a "rancher" service in cattle-system and use that if running an embedded K3s. 

https://github.com/rancher/rancher/issues/34015

### CODE REVIEW

**Review by brandond — DISMISSED** [review]

**Review by nickgerace — DISMISSED** [review]

**Comment by nickgerace** [comment]
Assigned committer @kinarashah. Need a committer review before merging

**Review by nickgerace — APPROVED** [review]

**Review by kinarashah — APPROVED** [review]

**Pull request merged**
